docs/validate-package.md

```diff
@@ -0,0 +1,90 @@
+
+## Linting
+
+As a quick check to statically verify that your package is valid, you can use the `operator-courier` project to check a few basic things:
+
+```
+$ pip3 install operator-courier
+$ operator-courier verify manifests/my-operator-package
+```
+
+This will check the packaging format for some basic checks to ensure that your package is set up correctly.
+
+You can also check for some additional requirements to ensure that your operator will display properly on [OperatorHub.io](https://operatorhub.io/)
+
+```
+$ operator-courier verify --ui_validate_io manifests/my-operator-package
+```
+
+## Add your package to a catalog
+
+The [operator-registry project](https://github.com/operator-framework/operator-registry) defines a format for storing sets of operators and exposing them to make them available on a cluster. As part of adding your package to an operator-registry catalog, the operator-registry tools will verify that your operator is packaged properly ("Does it have a valid CSV of the correct format?", "Does my CRD properly reference my CSVs?", etc.). The simplest way to test that your package can be added to a catalog is by actually attempting to create a catalog that includes your operator.
+
+To create a catalog that includes your package, simply build a container image that uses the operator-registry command line tools to generate a registry and serve it. For example, create a file in the root of your project called `registry.Dockerfile`
+
+```Dockerfile
+FROM quay.io/operator-framework/upstream-registry-builder as builder
+
+COPY manifests manifests
+RUN ./bin/initializer -o ./bundles.db
+
+FROM scratch
+COPY --from=builder /build/bundles.db /bundles.db
+COPY --from=builder /build/bin/registry-server /registry-server
+COPY --from=builder /bin/grpc_health_probe /bin/grpc_health_probe
+EXPOSE 50051
+ENTRYPOINT ["/registry-server"]
+CMD ["--database", "bundles.db"]
+```
+
+This Dockerfile assumes that your package is in a directory called `./manifests/` similar to [this example](https://github.com/operator-framework/operator-registry/tree/master/manifests). It copies your manifests into the builder image, runs `initializer`, then copies the output into the final scratch image and defines the run command to serve the operator-registry.
+
+Then just use your favorite container tooling to build the container image and push it to a registry:
+
+```
+docker build -t example-registry:latest -f registry-Dockerfile .
+docker push example-registry:latest
+```
+
+Your catalog is published and we are ready to use it on your cluster.
+
+## Install your operator
+
+Now that you have created an operator-registry image that hosts your operator's package, add that catalog to your cluster:
```

Comment:
There are also some other checks we can do to ensure a bundle is queryable (i.e. run catalog container and grpcurl its API).

Reply:
That seems like a lot of overhead to test if your package installs correctly, no? If OLM installs the operator it would imply that the API is healthy and responding properly.

Reply:
I agree that it will catch errors, but slower. To me, this seems like a classic [e2e vs unit problem](https://martinfowler.com/articles/practical-test-pyramid.html#TheTestPyramid). I was under the impression that this doc would include both coarse and fine grained catalog validations. The latter being useful for less apparent issues, like ensuring an update path includes a specific CSV w/o requiring a cluster to run a full set of updates.